apps/test/unit/templates/courseOverview/CourseEditorTest.js

```diff
@@ -0,0 +1,52 @@
+import {assert} from '../../../util/reconfiguredChai';
+import React from 'react';
+import {mount} from 'enzyme';
+import CourseEditor from '@cdo/apps/templates/courseOverview/CourseEditor';
+import {
+  stubRedux,
+  restoreRedux,
+  getStore,
+  registerReducers
+} from '@cdo/apps/redux';
+import teacherSections from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import {Provider} from 'react-redux';
+
+const defaultProps = {
+  name: 'csp',
+  title: 'Computer Science Principles 2017',
+  familyName: 'CSP',
+  versionYear: '2017',
+  descriptionShort: 'Desc here',
+  descriptionStudent: 'Desc here',
+  descriptionTeacher: 'Desc here',
+  scriptsInCourse: ['CSP Unit 1', 'CSP Unit 2'],
+  scriptNames: ['CSP Unit 1', 'CSP Unit 2'],
+  teacherResources: [],
+  hasVerifiedResources: false,
+  courseFamilies: ['CSP', 'CSD', 'CSF'],
+  versionYearOptions: ['2017', '2018', '2019']
+};
+
+describe('CourseEditor', () => {
+  beforeEach(() => {
+    stubRedux();
+    registerReducers({teacherSections});
+  });
+
+  afterEach(() => {
+    restoreRedux();
+  });
+
+  it('renders full course editor page', () => {
+    const wrapper = mount(
+      <Provider store={getStore()}>
+        <CourseEditor {...defaultProps} />
```

Comment:
Awesome work @dmcavoy ! I was not aware of this pattern of using `Provider` in unit tests and have always extracted an "unconnected" component and tested that by explicitly passing in the props it needs, which does not look like it would have worked here. This seems like a great way of ensuring that the page continues to render, and a great pattern for us to follow for other top-level components on other pages as a sort of mini integration test.

Reply:
I found the pattern in https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/templates/instructions/InstructionsCSFTest.jsx